### ISSUE TITLE
[question] correct usage of build_policy = "never"

### ISSUE THREAD

**Issue by volviq** [conan-io/conan#10991]
Conan version 1.44.0 OS: Windows 11

We created a conanfile.py which only contains a `package` step to be created with `conan export-pkg` instead of `conan create`.
As it doesn't use the `exports_sources` to copy files into the conan cache, I assumed it makes sense to disable the possibility 
to rebuild a package.
Therefore we set the
`build_policy = "never"`
So that it isn't possibly to attempt to recreate the conan package only from the conanfile, as this wouldn't work anyway.
Then I get the error message:
```ERROR: No package matching '[package name]' pattern found.```
this message also appears with conan create.

So my questions:
* is there anything I need to do, to get rid of this message, but keeping the `build_policy = "never"`
* Is my understanding correct of the purpose of the `build_policy = "never"`? Does it make sense in this case?

- [X] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by czoido** [comment]
Hi @volviq,

About the first question, I can't reproduce the issue. The use of `build_policy="never"` [seems correct](https://docs.conan.io/en/latest/reference/commands/creator/export-pkg.html) if you are using `conan export-pkg`
Maybe you can try if a minimal example for a recipe works for you with conan export-pkg? Like this:

from conans import ConanFile, tools

```python
class HelloConan(ConanFile):
    name = "hello"
    version = "0.1"
    settings = "os", "compiler", "build_type", "arch"
    build_policy="never"

    def package(self):
        self.copy("*")

    def package_info(self):
        self.cpp_info.libs = tools.collect_libs(self)
```

The second question: the purpouse of `build_policy="never"` is to avoid that packages created with `conan export-pkg` are rebuilt from sources, so this policy will make them be skipped when you do a  `--build=*` or `--build` in the command line in a conan command.

**Comment by volviq** [comment]
This is really strange. I'm using the proposed example:
```
$ cat conanfile.py
""" a Conan Recipe """
from conans import ConanFile

class HelloConan(ConanFile):
    name = "hello"
    version = "0.1"
    settings = "os", "compiler", "build_type", "arch"
    build_policy="never"

    def package(self):
        self.copy("*")

    def package_info(self):
        self.cpp_info.libs = tools.collect_libs(self)
```

There isn't really anything useful to package in the folder, but I assume this isn't relevant here.
I still got the same message with my version. So installed a new version (Conan version 1.47.0), and still see the message:
```
/c/temp/test
$ /c/app/conan/conan/conan.exe  export-pkg . --profile=ghs --force
[HOOK - attribute_checker.py] pre_export(): WARN: Conanfile doesn't have 'url'. It is recommended to add it as attribute
[HOOK - attribute_checker.py] pre_export(): WARN: Conanfile doesn't have 'license'. It is recommended to add it as attribute
[HOOK - attribute_checker.py] pre_export(): WARN: Conanfile doesn't have 'description'. It is recommended to add it as attribute
Exporting package recipe
hello/0.1: The stored package has not changed
hello/0.1: Exported revision: 46f0bdd2b944f7c273ab2d89d5a4c500
ERROR: No package matching 'hello' pattern found.
Packaging to 3d72df7fd913f583dd434c50537b20270064aaaf
hello/0.1: Generating the package
hello/0.1: Package folder c:\conan_cache\hello\0.1\_\_\package\3d72df7fd913f583dd434c50537b20270064aaaf
hello/0.1: Calling package()
hello/0.1 package(): Packaged 1 '.py' file: conanfile.py
hello/0.1 package(): Packaged 1 '.txt' file: my_current_view.txt
hello/0.1: Package '3d72df7fd913f583dd434c50537b20270064aaaf' created
hello/0.1: Created package revision 37f78372fc8ae35c5a44610583f3c200
```

So, still, I'm working in Windows. I can do a quick check in Linux if it behaves differently there.
My profile looks like the following:
```
[settings]
os=RTAOS
os_build=Windows
arch=rh850
arch_build=x86_64
compiler=ghs
compiler.version=comp_201517_1fp
build_type=Release
[options]
[build_requires]
[env]
CONAN_V2_MODE=1
```

**Comment by volviq** [comment]
Another test in Linux WSL:
```
conan  export-pkg .
WARN: Remotes registry file missing, creating default one in /home/simon/.conan/remotes.json
Auto detecting your dev setup to initialize the default profile (/home/simon/.conan/profiles/default)
Found gcc 9
gcc>=5, using the major as version

************************* WARNING: GCC OLD ABI COMPATIBILITY ***********************

Conan detected a GCC version > 5 but has adjusted the 'compiler.libcxx' setting to
'libstdc++' for backwards compatibility.
Your compiler is likely using the new CXX11 ABI by default (libstdc++11).

If you want Conan to use the new ABI for the default profile, run:

    $ conan profile update settings.compiler.libcxx=libstdc++11 default

Or edit '/home/simon/.conan/profiles/default' and set compiler.libcxx=libstdc++11

************************************************************************************



Default settings
        os=Linux
        os_build=Linux
        arch=x86_64
        arch_build=x86_64
        compiler=gcc
        compiler.version=9
        compiler.libcxx=libstdc++
        build_type=Release
*** You can change them in /home/simon/.conan/profiles/default ***
*** Or override with -s compiler='other' -s ...s***


[HOOK - attribute_checker.py] pre_export(): WARN: Conanfile doesn't have 'url'. It is recommended to add it as attribute
[HOOK - attribute_checker.py] pre_export(): WARN: Conanfile doesn't have 'license'. It is recommended to add it as attribute
[HOOK - attribute_checker.py] pre_export(): WARN: Conanfile doesn't have 'description'. It is recommended to add it as attribute
Exporting package recipe
hello/0.1: A new conanfile.py version was exported
hello/0.1: Folder: /home/simon/.conan/data/hello/0.1/_/_/export
hello/0.1: Exported revision: 46f0bdd2b944f7c273ab2d89d5a4c500
ERROR: No package matching 'hello' pattern found.
Packaging to 2a30b7d6ea2202e5393ebda51c8729dd2162b9f8
hello/0.1: Generating the package
hello/0.1: Package folder /home/simon/.conan/data/hello/0.1/_/_/package/2a30b7d6ea2202e5393ebda51c8729dd2162b9f8
hello/0.1: Calling package()
hello/0.1 package(): Packaged 1 '.py' file: conanfile.py
hello/0.1 package(): Packaged 1 '.txt' file: my_current_view.txt
hello/0.1: Package '2a30b7d6ea2202e5393ebda51c8729dd2162b9f8' created
hello/0.1: Created package revision 7c0cc472c250224ab48b078db7a8988
```

**Comment by memsharded** [comment]
I have been able to reproduce it. Thanks for reporting, we will have a look.

At least, it seems the package is being created, it is just an error message.

**Comment by memsharded** [comment]
Ok, I have been able to track this. It is an unfortunate UX bug, due to sharing the core code with other commands. But, it has no effect at all, it is just an incorrect message, but the command is working fine, the package is correctly created, etc.

I have checked it, and this is already fixed in 2.0, and it would be quite dirty to try to fix it in 1.X, so I vote to consider this bug closed in 2.0.
